### ISSUE TITLE
add a version "state" property.

### ISSUE THREAD

**Issue by Claudenw** [ewilderj/doap#80]
**Motivation**

The current DOAP lists versions but has no indication which versions are EOL, or bug fix only, or active.  For purposes of automating warnings about old dependencies such a tag would be extremely valuable.

**Feature**
Simply a new property with a domain of Version.  I am open to discussion of range and whether or not this should be a controlled vocabulary.  Personally I think that it should be a controlled vocabulary to make it easier to process the data automatically.

**Alternatives**
Property defined in an external vocabulary.
Property with free form text.

**Additional context**

**Comment by ewilderj** [comment]
This is indeed interesting. One approach would be to provide suggested vocabulary terms but be open to implementors also adding their own as appropriate. Do you have any examples where we could figure out the terms?

Possible terms that cross my mind immediately would include:

* experimental
* release candidate
* stable release
* development release
* long term supported
* deprecated

**Comment by Claudenw** [comment]
Does `deprecated` == `EOL` == `not supported`?

What is the difference between `development release` and `experimental`

**Comment by Claudenw** [comment]
The DOAP file might looks something like.

```
<!-- my-project-0.2.rdf, also symlinked to my-project.rdf -->
<Project xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns="http://usefulinc.com/ns/doap#">
    <name>my-project</name>
    <!-- … -->
    <canonical-release rdf:resource="./my-project.rdf"/>
    <release>
        <Version>
            <revision>0.2</revision>
            <state>stable release</state>
        </Version>
    </release>
    <release>
        <Version>
            <revision>master</revision>
            <other-release rdf:resource="./my-project-next.rdf"/>
            <state>development release</state>
        </Version>
    </release>
    <release>
        <Version>
            <revision>0.1</revision>
            <other-release rdf:resource="./my-project-0.1.rdf"/>
            <state>EOL</state>
        </Version>
    </release>
</Project>
```

**Comment by mwild1** [comment]
A good precedent here is [endoflife.date](https://endoflife.date). Their project is JSON based, and they have this schema: https://github.com/endoflife-date/endoflife.date/blob/04248046e72df040fee6f3d77644952b09321f85/api_v1/openapi.yml#L514

It's not exactly standard, but they've put quite a bit of effort into developing it, and have a lot of existing projects already, demonstrating that it at least works.

**Comment by webketje** [comment]
publiccode.yml initiative defines values `concept, development, beta, stable, obsolete`: https://yml.publiccode.tools/schema.core.html#key-developmentstatus

**Comment by kjetilk** [comment]
Yes, it sounds like something that would be useful. I like the idea of concept, development, beta and stable, but obsolete strikes me as odd, because you would probably not update the data unless there was good reason to, and everything is eventually obsoleted. So, I think probably deprecated is better, since that suggests an active act. And lts (for "long term support") sounds good too. I think we should have a URI for each of them though
